### PR TITLE
restructure

### PR DESCRIPTION
**Pull request by whooo**
Regarding https://github.com/tpm2-software/tpm2-pytss/issues/248

- [X] Move friendly ints to constants.py
- [X] Move generate_seed, hmac and encrypt from makecred to crypto
- [X] Move crypto to internal
- [X] move current utils to internal
- [X] Move make_credential and wrap from makecred to utils
- [X] Use constants in crypto instead of lib

Fixes https://github.com/tpm2-software/tpm2-pytss/issues/248 and https://github.com/tpm2-software/tpm2-pytss/issues/242

### CODE REVIEW

**Comment by whooo** [comment]
@joholl, could you have a look at https://github.com/tpm2-software/tpm2-pytss/pull/251/commits/62756dee4283fec91bff0fc8365c38df1771ef43 and see if it makes sense? the moved code does mostly seem to be used internally but I might have missed something.

File: tpm2_pytss/internal/crypto.py [hunk 1]
```diff
@@ -345,34 +348,100 @@ def verify_signature_hmac(signature, key, data):
...
+    if halg is None:
+        raise ValueError(f"unsupported digest algorithm {hashAlg}")
+    seed = secrets.token_bytes(halg.digest_size)
+    mgf = padding.MGF1(halg())
+    padd = padding.OAEP(mgf, halg(), label)
+    enc_seed = key.encrypt(seed, padd)
+    return (seed, enc_seed)
+
+
+def generate_ecc_seed(
```
Comment (williamcroberts) [inline tpm2_pytss/internal/crypto.py:397]: Why are the 2 seed functions non-underscore versus underscore?

Is your convention here that things under internal are all for internal use, but functions/methods that start with an `_` are not general internal API. Ie something like underscore prefix functions are static functions for use in file scope and non-underscore prefix internal API is for general internal API consumption?

**Review by williamcroberts — APPROVED** [review]
Generally LGTM, just a question. Can you rebase and update?

File: tpm2_pytss/internal/crypto.py (hunk 1 above)
Comment (whooo) [reply tpm2_pytss/internal/crypto.py:397]: I skipped them as they aren't imported anywhere, they are only called by _generate_seed.
I'll fix it when I rebase

**Pull request merged**